pkg/assets/assets.go

```diff
@@ -43,7 +43,20 @@ func New(dir string, data interface{}, predicates ...FileInfoPredicate) (Assets,
 	for path, bs := range files {
 		a, err := assetFromTemplate(path, bs, data)
 		if err != nil {
-			errs = append(errs, fmt.Errorf("failed to render %q: %v", path, err))
+			return nil, fmt.Errorf("failed to render %q: %v", path, err)
+		}
+
+		skipManifest := false
+		for _, manifestPredicate := range manifestPredicates {
+			shouldInclude, err := manifestPredicate(a.Data)
+			if err != nil {
+				return nil, fmt.Errorf("failed to check manifest filter %q: %v", path, err)
+			}
+			if !shouldInclude {
+				skipManifest = true
```

Comment:
Consider `break` here, subsequent predicates will not change `skipManifest` again.